src/overlays/actors/ovl_Fuusen/ac_fuusen.c

```diff
@@ -1,55 +1,464 @@
+                   (this->actor.world.pos.z <= 960.0f) && (this->actor.world.pos.z >= 800.0f)) {
+            this->escapeTimer = 777;
+            aFSN_setupAction(this, FSN_PROCESS_ESCAPE, game_play);
+            return;
+        }
+    }
+    this->windPower = mEnv_GetWindPowerF();
+    this->heightAmplitudeCounter += 500;
+    this->actor.speed = this->windPower * 0.5f + 1.0f;
+    add_calc(&this->actor.world.pos.y, (sin_s(this->heightAmplitudeCounter) * 10.0f) + balloonGroundY, 0.3f, 1.0f,
+             0.0f);
+    Game_play_Projection_Trans(game_play, &this->actor.world.pos, &screenPos);
+    if (!(screenPos.x < -40.0f) && !(screenPos.x > 360.0f) && !(screenPos.y < -40.0f) && !(screenPos.y > 280.0f)) {
+        mCoBG_BgCheckControll(NULL, &this->actor, 12.0f, 0, 0, 0, 0);
+        if (this->actor.colResult.hitWall & 1 || this->actor.colResult.unk1 & 1) {
+            this->heightOffset += 0.01f;
+            if (this->heightOffset >= 300.0f) {
+                this->heightOffset = 300.0f;
+            }
+        } else {
+            if (this->heightOffset > 110.0f) {
+                this->heightOffset -= 0.01f;
+            }
+        }
+        unitFG = mFI_GetUnitFG(this->actor.world.pos);
+        if ((unitFG != NULL) &&
+            ((*unitFG == 0x804) || (*unitFG == 0x809) || (*unitFG == 0x811) || (*unitFG == 0x819) ||
+             (*unitFG == 0x821) || (*unitFG == 0x829) || (*unitFG == 0x80A) || (*unitFG == 0x812) ||
+             (*unitFG == 0x81A) || (*unitFG == 0x822) || (*unitFG == 0x82A) || (*unitFG == 0x80B) ||
+             (*unitFG == 0x813) || (*unitFG == 0x81B) || (*unitFG == 0x823) || (*unitFG == 0x82B))) {
+            mFI_Wpos2UtCenterWpos(&screenPos, this->actor.world.pos);
+            screenPos.x -= 2.5f;
+            screenPos.y = mCoBG_GetBgY_OnlyCenter_FromWpos2(this->actor.world.pos, 0.0f) + 97.5f;
+            screenPos.z += 7.5f;
+            distX = screenPos.x - this->actor.world.pos.x;
+            distY = screenPos.y - this->actor.world.pos.y;
+            distZ = screenPos.z - this->actor.world.pos.z;
+            if ((sqrtf(SQ(distX) + SQ(distZ)) < 15.0f) && (sqrtf(SQ(distY)) < 15.0f)) {
+                aFSN_setupAction(this, FSN_PROCESS_WOOD_STOP, game_play);
+            }
+        } else {
+            if (!this->unk_1A4) {
+            // FAKE
+            dummy_label:
+                for (i = 0; i < ARRAY_COUNT(senkou_check_data); i++) {
+                    xyz_t_move(&screenPos, &this->actor.world.pos);
+                    yRot = senkou_check_data[i].z + this->actor.world.rot.y;
+                    screenPos.x += sin_s(yRot) * 80.0f;
+                    screenPos.z += cos_s(yRot) * 80.0f;
+                    unitFG = mFI_GetUnitFG(screenPos);
+                    if ((unitFG != NULL) &&
+                        ((*unitFG == 0x804) || (*unitFG == 0x809) || (*unitFG == 0x811) || (*unitFG == 0x819) ||
```

Comment:
These could use the defines from `m_name_table`, but it's a bit inconsistent throughout the repo at the moment, so dunno

Reply:
oh nice, will use what we currently have here so its better for search replacing